### PR TITLE
Cdd 1719 code conventions

### PR DESCRIPTION
**Pull request by cichockimc**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -14,16 +14,29 @@
...
-addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "0.10.0")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.12")
 
-addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "1.1.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-settings" % "3.5.0")
 
-addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.5.12")
+addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "1.0.0")
+
+addSbtPlugin("uk.gov.hmrc" % "sbt-git-stamp" % "5.3.0")
```
Comment (paulblakehmrc) [inline project/plugins.sbt]: latest is `5.5.0` - worth checking the rest of the plugins to ensure all are at the latest

File: test/component/ComponentSpec.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.scalatest._
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+
+abstract class ComponentSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest
+  with BeforeAndAfterEach with BeforeAndAfterAll {
+
+  private val externalServicesHost = "localhost"
```
Comment (paulblakehmrc) [inline test/component/ComponentSpec.scala]: both of these could be imported from the values set in `customs-api-common`

File: build.sbt [hunk 3]
```diff
@@ -14,82 +14,122 @@
...
+)
+
+lazy val playPublishingSettings: Seq[sbt.Setting[_]] = sbtrelease.ReleasePlugin.releaseSettings ++
+  Seq(credentials += SbtCredentials) ++
+  publishAllArtefacts
+
+lazy val scoverageSettings: Seq[Setting[_]] = Seq(
+  coverageExcludedPackages := "<empty>;com.kenshoo.play.metrics.*;.*definition.*;prod.*;testOnlyDoNotUseInAppConf.*;app.*;uk.gov.hmrc.BuildInfo;views.*;uk.gov.hmrc.apinotificationpull.config.*",
+  coverageMinimum := 95,
+  coverageFailOnMinimum := false,
```
Comment (paulblakehmrc) [inline build.sbt]: value should be `true`

File: test/unit/connectors/ApiNotificationQueueConnectorSpec.scala [hunk 4]
```diff
@@ -172,8 +170,9 @@ class ApiNotificationQueueConnectorSpec extends UnitSpec with ScalaFutures with
 
       await(connector.delete(notification))
 
-
-      wverify(deleteRequestedFor(url))
+      eventually {
```
Comment (paulblakehmrc) [inline test/unit/connectors/ApiNotificationQueueConnectorSpec.scala]: Consider using our WireMockRunner trait? And use port/host from `customs-api-common`?

File: app/uk/gov/hmrc/apinotificationpull/connectors/ServiceLocatorConnector.scala [hunk 5]
```diff
@@ -34,14 +33,14 @@ object Registration {
   implicit val regFormat = Json.format[Registration]
 }
 
-class ServiceLocatorConnector @Inject()(http: HttpClient, configuration: ServiceConfiguration) {
+class ServiceLocatorConnector @Inject()(http: HttpClient, configuration: ServiceConfiguration, logger: CdsLogger) {
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/apinotificationpull/connectors/ServiceLocatorConnector.scala]: Can we use the `ServiceLocatorConnector` in `customs-api-common`?

File: build.sbt [hunk 6]
```diff
@@ -8,14 +8,13 @@
  *     http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
+ * distributed under the License is distributed on an "ASbootStrapPlay IS" BASIS,
```
Comment (paulblakehmrc) [inline build.sbt]: typo

File: project/AppDependencies.scala [hunk 7]
```diff
@@ -3,17 +3,15 @@ import sbt._
 
 object AppDependencies {
 
-  private val customsApiCommonVersion = "1.27.0"
-  private val bootStrapPlayVersion = "1.7.0"
+  private val customsApiCommonVersion = "1.28.0"
   private val hmrcTestVersion = "3.0.0"
   private val scalaTestVersion = "3.0.5"
   private val pegDownVersion = "1.6.0"
   private val mockitoVersion = "2.18.3"
```
Comment (paulblakehmrc) [inline project/AppDependencies.scala]: turns out latest is 2.22.0

File: test/component/ComponentSpec.scala [hunk 8]
```diff
@@ -20,13 +20,14 @@ import org.scalatest._
 import org.scalatestplus.play.guice.GuiceOneAppPerTest
 import play.api.Application
 import play.api.inject.guice.GuiceApplicationBuilder
+import util.ExternalServicesConfig.{Host, Port}
 
 abstract class ComponentSpec extends FeatureSpec with GivenWhenThen with Matchers with GuiceOneAppPerTest
   with BeforeAndAfterEach with BeforeAndAfterAll {
 
   override def newAppForTest(testData: TestData): Application = new GuiceApplicationBuilder().configure(Map(
```
Comment (paulblakehmrc) [inline test/component/ComponentSpec.scala]: is this used anywhere?

File: test/unit/controllers/ServiceLocatorConnectorSpec.scala [hunk 9]
```diff
@@ -16,107 +16,83 @@
...
+      ).thenReturn(Future.failed(serviceLocatorException))
+
+      await(connector.register) shouldBe false
+
+      verify(mockHttpClient).POST(ameq("https://SERVICE-LOCATOR-HOST/registration"), ameq(registration),
+        ameq(Seq(CONTENT_TYPE -> JSON)))(any[Writes[Registration]], any[HttpReads[HttpResponse]], any[HeaderCarrier], any[MdcLoggingExecutionContext])
       verify(connector.handlerOK, never).apply()
       verify(connector.handlerError).apply(serviceLocatorException)
     }
+
```
Comment (paulblakehmrc) [inline test/unit/controllers/ServiceLocatorConnectorSpec.scala]: do we need this spec as api-common already has this?

File: test/component/ComponentSpec.scala (hunk 8 above)
Comment (cichockimc) [reply test/component/ComponentSpec.scala]: https://github.com/hmrc/api-notification-pull/blob/CDD-1719-code-conventions/test/component/ComponentSpec.scala#L30

https://github.com/hmrc/api-notification-pull/blob/CDD-1719-code-conventions/test/component/ComponentSpec.scala#L31

**Pull request merged**